tool/tsh/tsh.go

```diff
@@ -343,21 +343,39 @@ func onLogin(cf *CLIConf) {
 func onLogout(cf *CLIConf) {
 	client.UnlinkCurrentProfile()
 
-	// logout from all
-	if cf.Proxy == "" {
-		client.LogoutFromEverywhere(cf.Username)
-	} else {
-		tc, err := makeClient(cf, true)
+	// get access to the file system key store in ~/.tsh
+	flka, err := client.NewFSLocalKeyStore(client.FullProfilePath(""))
+	if err != nil {
+		fmt.Printf("Unable to logout user: %v\n", err)
+		return
+	}
+
+	// extract the proxy name
+	proxyHost, _, err := net.SplitHostPort(cf.Proxy)
+	if err != nil {
+		proxyHost = cf.Proxy
+	}
+
+	switch {
+	// proxy and username for key to remove
+	case proxyHost != "" && cf.Username != "":
+		err = flka.DeleteKey(proxyHost, cf.Username)
 		if err != nil {
-			utils.FatalError(err)
+			fmt.Printf("Unable to logout %v from %v: %v\n", cf.Username, proxyHost, err)
+			return
 		}
-		if err = tc.Logout(); err != nil {
-			if trace.IsNotFound(err) {
-				utils.FatalError(trace.Errorf("you are not logged into proxy '%s'", cf.Proxy))
-			}
-			utils.FatalError(err)
+		fmt.Printf("Logged out %v from %v.\n", cf.Username, proxyHost)
+	// remove all keys
+	case proxyHost == "" && cf.Username == "":
+		err = flka.DeleteKeys()
```

Comment:
check NotFound condition here - tsh logout called twice should not result in error

Reply:
`DeleteKeys` does not return an error. However, I cleaned up the error message when logging out a specific user a few lines up.